lib/Immediate/Immediate.cpp

```diff
@@ -333,6 +333,10 @@ int swift::RunImmediately(CompilerInstance &CI, const ProcessCmdLine &CmdLine,
 
   auto Result = (*JIT)->runMain(CmdLine);
 
+  // It is not safe to unmap memory that has been registered with the swift or
+  // objc runtime. Currently the best way to avoid that is to leak the JIT.
+  (void)JIT->release();
+
```

Comment:
Might be worth a FIXME to replace this with `detach` once that operation is implemented (https://github.com/llvm/llvm-project/issues/56714).